### ISSUE TITLE
Return oauth_secret_token for OAuth1.0 providers

### ISSUE THREAD

**Issue by kangmingtay** [supabase/gotrue#105]
# Feature request
Currently, only the provider_token (oauth_token) is being returned for OAuth1.0 providers. In the future, we would either want to return the `oauth_secret_token` as well or not at all. 

## Is your feature request related to a problem? Please describe.
Returning both the `provider_token` and `oauth_secret_token` will allow the service to retrieve a user's credentials in the future without having to ask the user for permission again. In the case of Twitter, the service can retrieve a user's tweets / retweet count / etc... 

## Describe the solution you'd like
OAuth1.0 tokens generally have a longer lasting lifetime compared to OAuth2.0 (Twitter's tokens do not expire but can be revoked by the user whenever). We can send the `oauth_secret_token` as part of the query parameters returned together with the `provider_token`. 

## Additional context
1. Is it secure / a good practice to send the `oauth_secret_token` as a query parameter?

**Comment by rishimohan** [comment]
+1 for this one! If it works out I can use it on kizie.co

**Comment by rishimohan** [comment]
@kangmingtay Hi Kang, any ETA/plans for this one? I've been facing auth issues in my existing setup and would really want to use Supabase before I get too many users in the app(it's already on production 😅)

**Comment by mike-wax** [comment]
@kangmingtay @kiwicopple Can we get an update on this one? The `provider_token` on it's own is not too helpful for Twitter.

**Comment by kangmingtay** [comment]
> @kangmingtay @kiwicopple Can we get an update on this one? The `provider_token` on it's own is not too helpful for Twitter.

Hey @mike-wax, we're currently in the midst of refactoring gotrue so that we can securely return the provider_token & provider_secret token (for twitter's case). We didn't support it initially because the supabase-js client library stores tokens in local storage. This is a security concern for twitter as twitter uses oauth1.0 - which returns tokens that are long-lived. If an attacker manages to get the tokens from local storage, the tokens do not expire and only the user can revoke them.

**Comment by mike-wax** [comment]
@kangmingtay thanks! Are you planning to store the secrets in supabase encrypted?

**Comment by rishimohan** [comment]
@kangmingtay Makes sense, looking forward to it!

**Comment by kangmingtay** [comment]
> @kangmingtay thanks! Are you planning to store the secrets in supabase encrypted?

Nope, for oauth1.0 (twitter), we don't intend to store it on the browser at all. We'll be returning the provider_token & provider_secret_token in the api response. For oauth2.0, the provider_token will be returned the same way as mentioned above and the provider_refresh_token will be stored as a http-only cookie. We'll be providing an endpoint to refresh your provider_token as well.

**Comment by Timonzimm** [comment]
Any update?

**Comment by hf** [comment]
Given that this was implemented in #490, I'm closing the issue. Feel free to re-open it.

**Comment by rishimohan** [comment]
@hf can you please share link to the PR where it's implemented? thanks

**Comment by hf** [comment]
@rishimohan I'll reopen the issue as this was only added to OAuth2.0 providers, while Twitter uses OAuth1.0.

**Comment by Timonzimm** [comment]
@kangmingtay any update? This is quite a blocking issue.

**Comment by Diogovski** [comment]
Months have passed, yet an update to be made...

**Comment by feresr** [comment]
+1 would love an update on this!

**Comment by operatorultra** [comment]
> Return oauth_secret_token for OAuth1.0 providers 

Will this be implemented? 

Otherwise will look at different solutions.

Thank you.

**Comment by jetlej** [comment]
+1

**Comment by Mumbo-Jumbo-3** [comment]
pls

**Comment by 019ec6e2** [comment]
This, plus 30% of failed logins on Twitter with absent email, is why I cannot use Supabase Auth anymore. Since authentication is a significant part of the whole stack, I would likely need to move away from Supabase altogether.

**Comment by Stan015** [comment]
Seems there's no update coming soon in regards to this issue. It's been open since 2021! I feel frustrated because it will cost me to switch from existing auth setup.
